fix(ui): Update bindings in Project Policy Config

vg006 opened the pull request:
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #22688

# Summary
This PR fixes the improper bindings of the attribute `disabled` in the [`project-policy-config.component.html`](https://github.com/goharbor/harbor/tree/main/src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html) component file. 

## Overview of changes made
- `project-policy-config.component.html`

  > Fixed all the improper bindings of the attribute `disabled` in the component, as described in the issue.
    ```diff
     <input 
    -   disabled="!allowUpdateProxyCacheConfiguration"
    +   [disabled]="!allowUpdateProxyCacheConfiguration"
      ...
     />
    ```
- `project-policy-config.component.ts`
   
  > Added logic to change the property `allowUpdateProxyCacheConfiguration` based on the permissions, since the variable is initialized with a default value `false` [(ref)](https://github.com/goharbor/harbor/tree/main/src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.ts#L107) in the file and the variable is neither used nor mutated.
    ```diff
     private getPermission(): void {
        this.userPermission
            .getPermission(
                this.projectId,
                USERSTATICPERMISSION.CONFIGURATION.KEY,
                USERSTATICPERMISSION.CONFIGURATION.VALUE.UPDATE
            )
            .subscribe(permissins => {
                this.hasChangeConfigRole = permissins as boolean;
    +           this.allowUpdateProxyCacheConfiguration =
    +             this.hasChangeConfigRole && !this.isProxyCacheProject;
            });
    }
    ```
## Result:
<img width="1556" height="126" alt="image" src="https://github.com/user-attachments/assets/14f8ba7d-e1ad-4a96-91f7-ed20776bbec4" />

---
Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. `release-note/update`, `release-note/enhancement`
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

vg006 commented:
To the maintainers @bupd @Vad1mo @chlins @MinerYang,

This will enable only the `System Admin` to configure Proxy Cache settings, not even `Project Admin` can configure it. 
Should it be configured in such way or need any change in access model?

vg006 commented:
@bupd
This branch is rebased on top of upstream's main branch and is now ready for merge.

bupd reviewed (changes requested):
@vg006 Thanks for your contribution,

you cannot update a proxy cache to normal and vice versa so the checkbox should always be in disabled state.

vg006 commented:
@bupd
I have made changes according to your review. Additionally, I also updated the misleading `PROXY_CACHE_TOOLTIP` messages. 

I have rebased on `main` branch as well, So can you review the changes once again?

bupd reviewed (approved):
/lgtm

bupd reviewed (approved):
/lgtm

Vad1mo commented:
we should not have a language change if it is absolutely needed, since  the change has not impact.

bupd reviewed (approved):
Good, Nothing else needed 

It's clear, the translation changes are wording changes we need to make to proxy cache - Since the addition of configurable proxy cache registries.

bupd commented:
@reasonerjt @wy65701436 - feel free to merge this PR - we already did multiple rounds of reviews on this one.

vg006 commented:
@Vad1mo @reasonerjt @wy65701436 
This PR has approved changes and is ready to merge.

bupd committed:
fix(ui): update project policy proxy cache config

Signed-off-by: Prasanth Baskar <bupdprasanth@gmail.com>

bupd commented:
Congrats @vg006 on your first merged PR

vg006 commented:
Thank you so much for accepting my contribution 🙂
